packages/easy-hydrator/src/TypeCaster/ObjectTypeCaster.php

```diff
@@ -54,12 +54,9 @@ public function retype(
             return $this->createObject($className, $value, $classConstructorValuesResolver);
         }
 
-        $objects = [];
-        foreach ($value as $objectData) {
-            $objects[] = $this->createObject($className, $objectData, $classConstructorValuesResolver);
-        }
-
-        return $objects;
+        return array_map(function ($objectData) use ($className, $classConstructorValuesResolver) {
+            return $this->createObject($className, $objectData, $classConstructorValuesResolver);
+        }, $value);
```

Comment:
This looks super identical to me.

What's the behavior difference?

Reply:
original variant reset array keys

Reply:
second option is

```php
        $objects = [];
        foreach ($value as $key => $objectData) {
            $objects[$key] = $this->createObject($className, $objectData, $classConstructorValuesResolver);
        }
```

Reply:
I see :+1: Thanks for explaining

Reply:
I see the 2nd version might be better, as the knowledge is explicit.